Mention on a few pages that minlength/maxlength only work on user input (#34402)
mdn/content: files/en-us/web/html/constraint_validation/index.md

For a basic introduction to these concepts, with examples, see the [Form validation tutorial](/en-US/docs/Learn/Forms/Form_validation).

> > **Note:** HTML Constraint validation doesn't remove the need for validation on the _server side_. Even though far fewer invalid form requests are to be expected, invalid ones can still be sent in many ways:

Reviewer: ```suggestion
> **Note:** HTML Constraint validation doesn't remove the need for validation on the _server side_. Even though far fewer invalid form requests are to be expected, invalid ones can still be entered and submitted in many ways:
```

Author: The current one is better. "Sent" doesn't imply you have to do it through the form, which is consistent with one of the bullet points below.

Reviewer: Yeah, good point.

>

> > - By modifying HTML through the inspector

Reviewer: ```suggestion
> - By modifying HTML via the browser's developer tools.
```

> > - By hand-crafting a request without using the form

Reviewer: ```suggestion
> - By hand-crafting an HTTP request without using the form.
```

> > - By programmatically writing content to the form (certain constraint validations are _only run_ for user input and not if you set the value of an input in JavaScript)

Reviewer: ```suggestion
> - By programmatically writing content into the form (certain constraint validations are _only run_ for user input, and not if you set the value of a form field using JavaScript).
```

>

> > Therefore, you need to always also validate input constraints on the server side, in a way that is consistent with what is done on the client side.

Reviewer: ```suggestion
> Therefore, you should always validate form data on the server side, consistent with what is done on the client side.
```

>
> - If the [`novalidate`](/en-US/docs/Web/HTML/Element/form#novalidate) attribute is set on the {{ HTMLElement("form") }} element, interactive validation of the constraints doesn't happen.
> - Calling the `submit()` method on the [`HTMLFormElement`](/en-US/docs/Web/API/HTMLFormElement) interface doesn't trigger a constraint validation. In other words, this method sends the form data to the server even if it doesn't satisfy the constraints. Call the `click()` method on a submit button instead.

> > - The `minlength` and `maxlength` constraints are only checked when the input comes from the user. They are not checked even when explicitly calling `checkValidity()` or `reportValidity()` if the value is set programmatically.

Reviewer: ```suggestion
> - The `minlength` and `maxlength` constraints are only checked on user-provided input. They are not checked if a value is set programmatically, even when explicitly calling `checkValidity()` or `reportValidity()`.
```
